packages/base/src/3dview/mainview.tsx

```diff
@@ -619,9 +621,27 @@ export class MainView extends React.Component<IProps, IStates> {
             this._selectedMeshes.push(el as any as BasicMesh);
             el.material.color = SELECTED_MESH_COLOR;
             el.material.linewidth = SELECTED_LINEWIDTH;
+          } else {
+            // Apply objColor for non-selected meshes
+            if (objColor && el.material?.color) {
+              const newColor = new THREE.Color(objColor);
+              el.material.color = newColor;
+              el.userData.originalColor = newColor.clone();
+            }
           }
         });
         this._meshGroup?.add(meshGroup);
+        // Apply the objColor to the mainMesh
+        if (!selected && objColor && mainMesh?.material?.color) {
+          const selectedMesh = this._meshGroup?.getObjectByName(
+            mainMesh.name
+          ) as BasicMesh;
+          if (selectedMesh && selectedMesh.material?.color) {
+            const newColor = new THREE.Color(objColor);
+            selectedMesh.material.color = newColor;
+            selectedMesh.userData.originalColor = newColor.clone();
+          }
+        }
```

Comment:
Isn't it a duplication with what's just above? Is this some leftover from the old implementation we should remove?

Reply:
Yes, I shall remove it. Thanks for noticing this. it's from one of the tries I was doing to have color.

Reply:
Done in [`d952a76`](https://github.com/jupytercad/JupyterCAD/pull/397/commits/d952a760f8b8b0aba2c221ceaf7f9af659eb23c2)